src/services/idleActionService.js

```diff
@@ -0,0 +1,44 @@
+class IdleActionService {
+  constructor(conn) {
+    this.conn = conn;
+  }
+
+  setIdleStatus(heroId, type) {
+    if (Number.isNaN(Number(heroId))) return Promise.reject(new Error(400));
+    if (type !== ('rest' || 'scout' || 'train')) {
+      return Promise.reject(new Error(400));
+    }
+    const currentTime = new Date() / 1000;
+    let query = 'SELECT * FROM idleStatus WHERE heroId = ?;';
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [heroId, type], (err, row) => {
+        if (err) {
+          reject(new Error(500));
+          return;
+        }
+        resolve(row);
+      });
+    }).then((row) => {
+      let values;
+      return new Promise((resolve, reject) => {
+        if (row.length === 0) {
+          query =
+            'INSERT INTO idleStatus(heroId, type, timestamp) VALUES(?, ?, ?);';
+          values = [heroId, type, currentTime];
+        } else if (row[0].type !== type) {
+          query =
+            'UPDATE idleStatus SET type = ?, timestamp = ? WHERE heroId = ?;';
+          values = [type, currentTime, heroId];
+        } else {
+          reject(new Error(400));
```

Comment:
pls make this a successful return. you dont have to add additional logic, just dont return with an error.

Reply:
I changed it, but it meant to be an error for the scenario: when the user wants to select an idle action which is already running and in the database, so it was an error